src/irr/video/COpenGLRenderpassIndependentPipeline.h

```diff
@@ -12,7 +12,40 @@ namespace video
 class COpenGLRenderpassIndependentPipeline : public IGPURenderpassIndependentPipeline
 {
 public:
-    using IGPURenderpassIndependentPipeline::IGPURenderpassIndependentPipeline;
+    COpenGLRenderpassIndependentPipeline(
+        core::smart_refctd_ptr<IGPUPipelineLayout> _layout,
```

Comment:
when your function takes a smart refcounted pointer (especially in constructors) that you'll be keeping hold of, make functions that only take r-value references.

This stops ontructor/desctructor being called, an atomic operation and freezing of the core.

Reply:
yea, but forces caller to drop ownership :/

Reply:
not really, it forces the programmer to do an explicit copy
```
core::smart_refctd_ptr<IGPUPipelineLayout> pointerIWantToShare;
new COpenGLRenderpassIndependentPipeline(core::smart_refct_ptr(pointerIWantToShare),...);
```

Then when you see this in the code (Especially when you actually write it) it will painfully stand out that there will be a `LOCK` instruction, and an atomic integer operation will be carried out (otherwise a programmer unfamiliar with the engine could be blissfully ignorant and pass smart pointers by copy ctor or copy assignment thinking they have little cost).

Reply:
resolved